### PR TITLE
Ignore some tests on macOS and Windows

### PR DESCRIPTION
**Pull request by JohnTitor**
Some tests cause timeout on GitHub Actions (it's Actions specific problem? I'm not sure). So ignore them for now.
And I noticed vcpkg installation takes long time (8~10min), I'm investigating how we cache openssl.

### CODE REVIEW

File: tests/test_server.rs [hunk 1]
```diff
@@ -741,6 +741,7 @@ async fn test_brotli_encoding_large_openssl() {
 }
 
 #[cfg(all(feature = "rustls", feature = "openssl"))]
+#[cfg(not(any(target_os = "macos", windows)))]
```
Comment (fafhrd91) [inline tests/test_server.rs:744]: this fails on osx only

File: awc/tests/test_client.rs [hunk 2]
```diff
@@ -46,6 +46,7 @@ const STR: &str = "Hello World Hello World Hello World Hello World Hello World \
                    Hello World Hello World Hello World Hello World Hello World";
 
 #[actix_rt::test]
+#[cfg(not(windows))]
```
Comment (fafhrd91) [inline awc/tests/test_client.rs:49]: does it fail on windows?

File: actix-http/tests/test_rustls.rs [hunk 3]
```diff
@@ -121,6 +121,7 @@ async fn test_h2_body1() -> io::Result<()> {
     Ok(())
 }
 
+#[cfg(not(any(target_os = "macos", windows)))]
```
Comment (fafhrd91) [inline actix-http/tests/test_rustls.rs:124]: does it fail?

File: actix-http/tests/test_openssl.rs [hunk 4]
```diff
@@ -115,6 +115,7 @@ async fn test_h2_body() -> io::Result<()> {
     Ok(())
 }
 
+#[cfg(not(any(target_os = "macos", windows)))]
```
Comment (fafhrd91) [inline actix-http/tests/test_openssl.rs:118]: i do development on osx, so disabling osx tests breaks my process

**Comment by JohnTitor** [comment]
Their tests actually are failed during my debugging. But I noticed this approach isn't good (I didn't know you use macOS, sorry!) and we need more investigation. I'm closing this for now.

**Pull request closed without merging**